Userland/Libraries/LibSQL/AST/CreateTable.cpp

```diff
@@ -35,13 +35,17 @@ ResultOr<ResultSet> CreateTable::execute(ExecutionContext& context) const
             type = SQLType::Integer;
         else if (column.type_name()->name().is_one_of("FLOAT"sv, "NUMBER"sv))
             type = SQLType::Float;
+        else if (column.type_name()->name().is_one_of("BOOL"sv, "BOOLEAN"sv))
+            type = SQLType::Boolean;
         else
             return Result { SQLCommand::Create, SQLErrorCode::InvalidType, column.type_name()->name() };
 
         table_def->append_column(column.name(), type);
     }
 
     TRY(context.database->add_table(*table_def));
+    TRY(context.database->commit());
```

Comment:
> 1. I feel like this should result in a `FIXME` somewhere

For sure!

> 2. Can we remove the `commit` in `Database`'s destructor?

Sure - it's a no-op for the destructor if there's no pending writes, but this would get rid of that big comment block.

> 3. Instead of adding a `commit` in every statement, can't we add a single one to `Statement::execute(Database)`?

We can, we just have to manually do error handling in `Statement::execute`, but at least it leaves the transactional FIXME in one location.

Reply:
Just to be sure, you added it to `SQLStatement` instead of `Statement` and as such it now only works through `SQLServer`. Was that the intention?

Reply:
Ah I see, I forgot about that Statement overload. I'll move it there, lets us use `TRY` as well.